src/engine/sched.c

```diff
@@ -864,17 +876,20 @@ process_req(struct dss_xstream *dx, struct sched_request *req)
 	if (req->sr_attr.sra_flags & SCHED_REQ_FL_NO_DELAY)
 		goto kickoff;
 
-	for (i = SCHED_REQ_UPDATE; i < SCHED_REQ_MAX; i++)
-		ult_timeout += info->si_kicked_req_cnt[i] * req_latencys[i];
-	ult_timeout /= 1000;
-
 	/* Request expired */
-	if ((info->si_cur_ts - req->sr_enqueue_ts + ult_timeout) >
-	    (req->sr_attr.sra_timeout * 2 / 3))
+	if (req->sr_attr.sra_timeout &&
```

Comment:
Is it possible that 'sra_timeout > MAX_CYCLE_TIME'? It's better to change this check from 'sra_timeout != 0" to "sra_timeout > MAX_CYCLE_TIME".

Reply:
Though, MAX_CYCLE_TIME should be less than 1 seconds, and sra_timeout normally larger than 1 second, but check is better.